index.js

```diff
@@ -54,9 +55,17 @@ function upperCaseFirst(s) {
 }
 
 function createSwatch(name, color) {
-  const { r, g, b, a } = parseColor(color);
-  return `list.setColor(NSColor.init(red: ${r / 255}, green:${g /
+  if (/[0-9A-Fa-f]{6}/.test(color) || /[0-9A-Fa-f]{3}/.test(color)) {
```

Comment:
Checking the syntax of the color is a good idea. The regex could be a bit stricter though. With this implementation values like `aaah!` or `fafafafafaafafafafafa` or `happy halloweeen` are considered valid because the regex only checks if there is sequence of 3 or 6 hexadecimal characters somewhere in the string w/o looking at the string boundaries.

I would suggest to change this to:

```js
  if (/^#(?:[0-9A-Fa-f]{6}|[0-9A-Fa-f]{3})$/.test(color)) (
```

Let me know if I should explain what this does :-)

Reply:
Yes, agreed. That is better.